### PR TITLE
Add a shadow opacity property to Light3D

### PR DESCRIPTION
**Pull request by Calinou**
This can be used to make shadows translucent for a specific light. Performance seems to be identical from before, both with shadows enabled and disabled. This is a partial reimplementation of the `shadow_color` property, which had to be [removed for optimization reasons](https://github.com/godotengine/godot/pull/45023).

The light distance fade system also uses this to smoothly fade the shadow when the light fade transition distance is greater than 0.

### CODE REVIEW

File: servers/rendering/renderer_rd/shaders/scene_forward_lights_inc.glsl [hunk 1]
```diff
@@ -732,6 +733,7 @@ float light_process_spot_shadow(uint idx, vec3 vertex, vec3 normal) {
 				}
 
 				shadow /= float(sc_penumbra_shadow_samples);
+				shadow = shadow * spot_lights.data[idx].shadow_opacity + 1.0 - spot_lights.data[idx].shadow_opacity;
```
Comment (reduz) [inline servers/rendering/renderer_rd/shaders/scene_forward_lights_inc.glsl:736]: use mix(), this looks pretty weird.

**Review by reduz — DISMISSED** [review]

**Comment by reduz** [comment]
Seems ok save for the comment I made. Needs rebase.

File: servers/rendering/renderer_rd/shaders/scene_forward_lights_inc.glsl (hunk 1 above)
Comment (Calinou) [reply servers/rendering/renderer_rd/shaders/scene_forward_lights_inc.glsl]: I hadn't thought about using `mix()` here, it makes sense :slightly_smiling_face:

**Comment by Calinou** [comment]
Rebased (with suggestion applied) and tested again, it works as expected on all light types.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**